src/devices/README.md

```diff
@@ -251,6 +255,138 @@ iot/
+|---|---|
+| Always use [UnitsNet](https://github.com/angularsen/UnitsNet) when possible: | Don't implement you own: |
+| ```public void SetEnvironmentData(Temperature temperature, Ratio humidity)``` | ```public void SetEnvironmentData(float temperature, float humidity)``` |
+| Temperature is a classic one, you have lots of available units available, [check here](https://github.com/angularsen/UnitsNet/tree/master/UnitsNet/GeneratedCode/Units) if anything exists before starting|In this case, there are existing units, use the existing one |
+
+| Do | Don't |
+|---|---|
+| Decribe your unit if it does not exist in [UnitsNet](https://github.com/angularsen/UnitsNet)|Leave an unkonwn measurement element |
+| ```public bool TryGetMyUnit(out double myUnitInUnitDescription)``` | ```public bool TryGetMyUnit(out double myUnit)``` |
+| In this case, you need to speciy in the variable name the unit used. Be as precise as possible | Never leave any unit without description in the naming if it's not an support UnitsNet unit. |
+
+| Do | Don't |
+|---|---|
+| Always add range of the units in the XML description|Never leave a unit without a full range description|
+| ```/// <param name="equivalentCO2">The equivalent CO2 (eCO2) output range for CCS811 is from 400ppm up to 29206ppm.</param>``` | ```/// <param name="equivalentCO2">The equivalent CO2 (eCO2)</param>``` |
+| Even when using the UnitsNet, make sure you give the usual range of a sensor value. | This makes it difficult to understand  which range the sensor is using. |
+
+### Naming of functions
+
+| Do |
+|---|
+| When reading a sensor and outputting data, use ```Try``` for example: |
+| ```public bool TryGetTemperature(out Temperature temperature)``` |
+| This makes it easy to understand that if you read and the retruned value is ```false``` the reading is not correct. And this means as well this function should not raise an exception when trying to read the sensor data. |
+
+| Do|
+|---|
+| Use action verbs for any function that makes an action. |
+| ```public void SetEnvironmentData(Temperature temperature, Ratio humidity)``` |
+| In this example, the function Set the environement values of temperature and humidity. It makes it very easy to understand, on top of the XML comment what the function is doing. |
+
+| Do |
+|---|
+| Use properties for a single element Set and/or Get function |
+| ```public OperationMode OperationMode``` |
+| This function then can be implemented with a get and set: |
+
+```csharp
+public OperationMode OperationMode
+{
+    get
+    {
+        var mode = ReadRegister(Register.MEAS_MODE);
+        mode = (byte)((mode >> 4) & 0b0000_0111);
+        return (OperationMode)mode;
+    }
+
+    set
+    {
+        var mode = ReadRegister(Register.MEAS_MODE);
+        // Clear previous mode
+        mode = (byte)(mode & 0b1000_1111);
+        mode = (byte)(mode | (((byte)(value)) << 4));
+        WriteRegister(Register.MEAS_MODE, mode);
+    }
+}
+```
+
+| Do | Don't |
```

Comment:
I'd put this above the `Try...` rule and also add that it is acceptable to use abbreviation for commonly known abbreviations (i.e. Html/Xml)